### PR TITLE
Added federated tests for TypeScript target

### PR DESCRIPTION
**Pull request by suyourice**
TypeScript tests for federated execution: ChainWithDelay, TopLevelArtifacts, InternalDelay(lib)

### CODE REVIEW

File: test/TypeScript/src/federated/ChainWithDelay.lf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import InternalDelay from "../lib/InternalDelay.lf";
+import TestCount from "../lib/TestCount.lf";
+
+federated reactor {
+    c = new Count(period = 1 msec);
+    i = new InternalDelay(delay = 500 usec);
+    t = new TestCount(numInputs = 3);
+    c.out -> i.inp;
+    i.out -> t.inp;
+}
```
Comment (CloverCho) [inline test/TypeScript/src/federated/ChainWithDelay.lf:20]: I think you have to add new line at the end of code

File: test/TypeScript/src/lib/InternalDelay.lf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    input inp:number;
+    output out:number;
+    logical action d:number;
+    reaction(inp) -> d {=
+        actions.d.schedule(delay, inp as number);
+    =}
+    reaction(d) -> out {=
+        out = d;
+    =}
+}
```
Comment (CloverCho) [inline test/TypeScript/src/lib/InternalDelay.lf:17]: I think you have to add new line at the end of code

File: test/TypeScript/src/federated/TopLevelArtifacts.lf [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    tc = new TestCount();
+    c.out -> tc.inp;
+    
+    reaction (shutdown) {=
+        if (successes != 3) {
+            util.requestErrorStop(`Failed to properly execute top-level reactions`);
+        }
+        console.log(`SUCCESS!`);
+    =}
+}
```
Comment (CloverCho) [inline test/TypeScript/src/federated/TopLevelArtifacts.lf:43]: I think you have to add new line at the end of code

**Review by lhstrh — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**